### PR TITLE
Add the Orleans meters

### LINKED ISSUE #33984: Runtime monitoring - Available statistics and metrics not documented
It would be really useful if this doc detailed the statistics and metrics available from Orleans. Maybe a table of them?

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 6e987171-00a7-5cd4-8b0d-674e4e5415fb
* Version Independent ID: b75215a0-bb33-efe0-5eb4-8b0e1b35fa41
* Content: [Runtime monitoring and logs](https://learn.microsoft.com/en-us/dotnet/orleans/host/monitoring/)
* Content Source: [docs/orleans/host/monitoring/index.md](https://github.com/dotnet/docs/blob/main/docs/orleans/host/monitoring/index.md)
* Product: **dotnet-orleans**
* GitHub Login: @IEvangelist
* Microsoft Alias: **dapine**

---
[Associated WorkItem - 153651](https://dev.azure.com/msft-skilling/Content/_workitems/edit/153651)

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

Add the Orleans meters—need help from @ReubenBond for the descriptions of each.

Fixes #33984

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/diagnostics/compare-metric-apis.md](https://github.com/dotnet/docs/blob/56d8e8aa9dd3b9f08d1d24a67ab0b2b6305eea6a/docs/core/diagnostics/compare-metric-apis.md) | [Metric APIs comparison](https://review.learn.microsoft.com/en-us/dotnet/core/diagnostics/compare-metric-apis?branch=pr-en-us-36931) |
| [docs/orleans/host/monitoring/index.md](https://github.com/dotnet/docs/blob/56d8e8aa9dd3b9f08d1d24a67ab0b2b6305eea6a/docs/orleans/host/monitoring/index.md) | [Orleans observability](https://review.learn.microsoft.com/en-us/dotnet/orleans/host/monitoring/index?branch=pr-en-us-36931) |

### CODE REVIEW

File: docs/orleans/host/monitoring/index.md [hunk 1]
```diff
@@ -79,9 +79,245 @@ Press p to pause, r to resume, q to quit.
...
+A collection of catalog meters that are used to monitor the catalog layer.
+
+| Meter name | Type | Description |
+|--|--|--|
+| `orleans-catalog-activations` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the number of catalog activations. |
+| `orleans-catalog-activation-working-set` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the number of activations within the working set. |
+| `orleans-catalog-activation-created` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of created activations. |
+| `orleans-catalog-activation-destroyed` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of destroyed activations. |
+| `orleans-catalog-activation-failed-to-activate` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of activations that failed to activate. |
+| `orleans-catalog-activation-collections` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of activation collections. |
```
Comment (ReubenBond) [inline docs/orleans/host/monitoring/index.md:154]:
```suggestion
| `orleans-catalog-activation-collections` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of idle activation collections. |
```

File: docs/orleans/host/monitoring/index.md [hunk 2]
```diff
@@ -79,9 +79,245 @@ Press p to pause, r to resume, q to quit.
...
+| `orleans-directory-validations-cache-received` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of directory cache validations received. |
+| `orleans-directory-partition-size` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the directory partition size. |
+| `orleans-directory-cache-size` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the directory cache size. |
+| `orleans-directory-ring-size` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the directory ring size. |
+| `orleans-directory-ring-local-portion-distance` |<xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the directories local portion distance. |
+| `orleans-directory-ring-local-portion-percentage` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the directories local portion percentage. |
+| `orleans-directory-ring-local-portion-average-percentage` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the directories local average percentage. |
```
Comment (ReubenBond) [inline docs/orleans/host/monitoring/index.md:181]:
```suggestion
| `orleans-directory-ring-local-portion-distance` |<xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the ring range owned by the local directory partition. |
| `orleans-directory-ring-local-portion-percentage` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the ring range owned by the local directory, represented as a percentage of the total range. |
| `orleans-directory-ring-local-portion-average-percentage` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the average percentage of the directory ring range owned by each silo, giving a representation of how balanced directory ownership. |
```

File: docs/orleans/host/monitoring/index.md [hunk 3]
```diff
@@ -79,9 +79,245 @@ Press p to pause, r to resume, q to quit.
...
+| `orleans-directory-ring-local-portion-percentage` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the ring range owned by the local directory, represented as a percentage of the total range. |
+| `orleans-directory-ring-local-portion-average-percentage` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the average percentage of the directory ring range owned by each silo, giving a representation of how balanced directory ownership. |
+| `orleans-directory-registrations-single-act-issued` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of directory single act registrations issued. |
+| `orleans-directory-registrations-single-act-local` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of directory single act registrations local. |
+| `orleans-directory-registrations-single-act-remote-sent` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of directory single act registrations sent remotely. |
+| `orleans-directory-registrations-single-act-remote-received` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of directory single act registrations received remotely. |
```
Comment (ReubenBond) [inline docs/orleans/host/monitoring/index.md:185]:
```suggestion
| `orleans-directory-registrations-single-act-issued` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of directory single activation registrations issued. |
| `orleans-directory-registrations-single-act-local` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of directory single activation registrations handled by the local directory partition. |
| `orleans-directory-registrations-single-act-remote-sent` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of directory single activation registrations sent to a remote directory partition. |
| `orleans-directory-registrations-single-act-remote-received` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of directory single activation registrations received from remote hosts. |
```

**Review by BillWagner — APPROVED** [review]
I had a couple suggestions, and then this is ready to :shipit:

File: docs/orleans/host/monitoring/index.md [hunk 4]
```diff
@@ -79,9 +79,245 @@ Press p to pause, r to resume, q to quit.
 
 For more information, see [Investigate performance counters (dotnet-counters)](../../../core/diagnostics/dotnet-counters.md).
 
+### Orleans meters
+
+Orleans uses the [System.Diagnostics.Metrics](../../../core/diagnostics/compare-metric-apis.md#systemdiagnosticsmetrics) APIs to collect metrics. Orleans categorizes each meter into domain-centric concerns, such as networking, messaging, gateway, and so on. The following subsections describe the meters that Orleans uses.
+
+#### Networking
+
+A collection of networking meters that are used to monitor the networking layer.
```
Comment (IEvangelist) [inline docs/orleans/host/monitoring/index.md:88]:
```suggestion
The following table represents a collection of networking meters that are used to monitor the Orleans networking layer.
```

File: docs/orleans/host/monitoring/index.md [hunk 5]
```diff
@@ -79,9 +79,245 @@ Press p to pause, r to resume, q to quit.
...
+| `orleans-messaging-pings-reply-received` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of ping replies received. |
+| `orleans-messaging-pings-reply-missed` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of ping replies missed. |
+| `orleans-messaging-expired"` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of messages that have expired. |
+| `orleans-messaging-rejected` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of messages that have been rejected. |
+| `orleans-messaging-rerouted` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of messages that have been rerouted. |
+| `orleans-messaging-sent-local` | <xref:System.Diagnostics.Metrics.ObservableCounter%601> | An observable counter representing the number of local messages sent. |
+
+#### Gateway
+
+A collection of gateway meters that are used to monitor the gateway layer.
```
Comment (IEvangelist) [inline docs/orleans/host/monitoring/index.md:124]:
```suggestion
The following table represents a collection of gateway meters that are used to monitor the Orleans gateway layer.
```

File: docs/orleans/host/monitoring/index.md [hunk 6]
```diff
@@ -79,9 +79,245 @@ Press p to pause, r to resume, q to quit.
...
+| `orleans-catalog-activation-destroyed` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of destroyed activations. |
+| `orleans-catalog-activation-failed-to-activate` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of activations that failed to activate. |
+| `orleans-catalog-activation-collections` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of idle activation collections. |
+| `orleans-catalog-activation-shutdown` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of shutdown activations. |
+| `orleans-catalog-activation-non-existent` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of non-existent activations. |
+| `orleans-catalog-activation-concurrent-registration-attempts` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of concurrent activation registration attempts. |
+
+#### Directory
+
+A collection of directory meters that are used to monitor the directory layer.
```
Comment (IEvangelist) [inline docs/orleans/host/monitoring/index.md:161]:
```suggestion
The following table represents a collection of directory meters that are used to monitor the Orleans directory layer.
```

File: docs/orleans/host/monitoring/index.md [hunk 7]
```diff
@@ -79,9 +79,245 @@ Press p to pause, r to resume, q to quit.
...
+| `orleans-directory-unregistrations-remote-sent` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of directory deregistrations sent to remote directory partitions. |
+| `orleans-directory-unregistrations-remote-received` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of directory deregistrations received from remote hosts. |
+| `orleans-directory-unregistrations-many-issued` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of directory multi-activation deregistrations issued. |
+| `orleans-directory-unregistrations-many-remote-sent` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of directory multi-activations deregistrations sent to remote directory partitions. |
+| `orleans-directory-unregistrations-many-remote-received` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of directory multi-activation deregistrations received from remote hosts. |
+
+#### Consistent ring
+
+A collection of consistent ring meters that are used to monitor the consistent ring layer.
```
Comment (IEvangelist) [inline docs/orleans/host/monitoring/index.md:196]:
```suggestion
The following table represents a collection of consistent ring meters that are used to monitor the Orleans consistent ring layer.
```

File: docs/orleans/host/monitoring/index.md [hunk 8]
```diff
@@ -79,9 +79,245 @@ Press p to pause, r to resume, q to quit.
...
+
+| Meter name | Type | Description |
+|--|--|--|
+| `orleans-consistent-ring-size` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the consistent ring size. |
+| `orleans-consistent-ring-range-percentage-local` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the consistent ring local percentage. |
+| `orleans-consistent-ring-range-percentage-average` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the consistent ring average percentage. |
+
+#### Watchdog
+
+A collection of watchdog meters that are used to monitor the watchdog layer.
```
Comment (IEvangelist) [inline docs/orleans/host/monitoring/index.md:206]:
```suggestion
The following table represents a collection of watchdog meters that are used to monitor the Orleans watchdog layer.
```

File: docs/orleans/host/monitoring/index.md [hunk 9]
```diff
@@ -79,9 +79,245 @@ Press p to pause, r to resume, q to quit.
...
+
+| Meter name | Type | Description |
+|--|--|--|
+| `orleans-watchdog-health-checks` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of watchdog health checks. |
+| `orleans-watchdog-health-checks-failed` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of failed watchdog health checks. |
+
+#### Client
+
+A collection of client meters that are used to monitor the client layer.
```
Comment (IEvangelist) [inline docs/orleans/host/monitoring/index.md:215]:
```suggestion
The following table represents a collection of client meters that are used to monitor the Orleans client layer.
```

File: docs/orleans/host/monitoring/index.md [hunk 10]
```diff
@@ -79,9 +79,245 @@ Press p to pause, r to resume, q to quit.
...
+A collection of miscellaneous meters that are used to monitor various layers.
+
+| Meter name | Type | Description |
+|--|--|--|
+| `orleans-grains` | <xref:System.Diagnostics.Metrics.Counter%601> | A count representing the number of grains. |
+| `orleans-system-targets` | <xref:System.Diagnostics.Metrics.Counter%601> | A count representing the number of system targets. |
+
+#### App requests
+
+A collection of app request meters that are used to monitor the app request layer.
```
Comment (IEvangelist) [inline docs/orleans/host/monitoring/index.md:232]:
```suggestion
The following table represents a collection of app request meters that are used to monitor the Orleans app request layer.
```

File: docs/orleans/host/monitoring/index.md [hunk 11]
```diff
@@ -79,9 +79,245 @@ Press p to pause, r to resume, q to quit.
...
+
+| Meter name | Type | Description |
+|--|--|--|
+| `orleans-reminders-tardiness` | <xref:System.Diagnostics.Metrics.Histogram%601> | A histogram representing the number of seconds a reminder is tardy. |
+| `orleans-reminders-active` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the number active reminders. |
+| `orleans-reminders-ticks-delivered` | <xref:System.Diagnostics.Metrics.Counter%601> | A count representing the number of reminder ticks that have been delivered. |
+
+#### Storage
+
+A collection of storage meters that are used to monitor the storage layer.
```
Comment (IEvangelist) [inline docs/orleans/host/monitoring/index.md:251]:
```suggestion
The following table represents a collection of storage meters that are used to monitor the Orleans storage layer.
```

File: docs/orleans/host/monitoring/index.md [hunk 12]
```diff
@@ -79,9 +79,245 @@ Press p to pause, r to resume, q to quit.
...
+| `orleans-storage-read-errors` | <xref:System.Diagnostics.Metrics.Counter%601> |  A count representing the number of storage read errors. |
+| `orleans-storage-write-errors` | <xref:System.Diagnostics.Metrics.Counter%601> |  A count representing the number of storage write errors. |
+| `orleans-storage-clear-errors` | <xref:System.Diagnostics.Metrics.Counter%601> |  A count representing the number of storage clear errors. |
+| `orleans-storage-read-latency` | <xref:System.Diagnostics.Metrics.Histogram%601> | A histogram representing the storage read latency in milliseconds. |
+| `orleans-storage-write-latency` | <xref:System.Diagnostics.Metrics.Histogram%601> | A histogram representing the storage write latency in milliseconds. |
+| `orleans-storage-clear-latency` | <xref:System.Diagnostics.Metrics.Histogram%601> | A histogram representing the storage clear latency in milliseconds. |
+
+#### Streams
+
+A collection of stream meters that are used to monitor the stream layer.
```
Comment (IEvangelist) [inline docs/orleans/host/monitoring/index.md:264]:
```suggestion
The following table represents a collection of stream meters that are used to monitor the Orleans stream layer.
```

File: docs/orleans/host/monitoring/index.md [hunk 13]
```diff
@@ -79,9 +79,245 @@ Press p to pause, r to resume, q to quit.
...
+| `orleans-streams-queue-cache-oldest-to-newest-duration` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the duration from the oldest to the newest stream queue cache. |
+| `orleans-streams-queue-cache-oldest-age` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the age of the oldest cached message. |
+| `orleans-streams-queue-cache-pressure` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the pressure on the stream queue cache. |
+| `orleans-streams-queue-cache-under-pressure` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing whether the stream queue cache is under pressure. |
+| `orleans-streams-queue-cache-pressure-contribution-count` | <xref:System.Diagnostics.Metrics.ObservableCounter%601> | An observable counter representing the stream queue cache pressure contributions. |
+
+#### Transactions
+
+A collection of transaction meters that are used to monitor the transaction layer.
```
Comment (IEvangelist) [inline docs/orleans/host/monitoring/index.md:309]:
```suggestion
The following table represents a collection of transaction meters that are used to monitor the Orleans transaction layer.
```

**Pull request merged**